nixos/modules/services/mail/sympa.nix

```diff
@@ -0,0 +1,536 @@
+  };
+
+  mainConfig = pkgs.writeText "sympa.conf" ''
+    domain     ${if cfg.mainDomain != null then cfg.mainDomain else (head fqdns)}
+    listmaster ${concatStringsSep "," cfg.listMasters}
+    lang       ${cfg.lang}
+
+    home        ${dataDir}/list_data
+    arc_path    ${dataDir}/arc
+    bounce_path ${dataDir}/bounce
+
+    db_type ${cfg.database.type}
+    db_name ${cfg.database.name}
+    ${optionalString (cfg.database.host != null) "db_host ${cfg.database.host}"}
+    ${optionalString (cfg.database.port != null) "db_port ${cfg.database.port}"}
+    ${optionalString (cfg.database.user != null) "db_user ${cfg.database.user}"}
+    db_passwd #dbpass#
+
+    ${optionalString (cfg.mta.type == "postfix") ''
+      sendmail         /run/wrappers/bin/sendmail
+      sendmail_aliases ${dataDir}/sympa_transport
+
+      aliases_program ${pkgs.postfix}/bin/postmap
+      aliases_db_type hash
+    ''}
+
+    ${optionalString cfg.web.enable ''
+      static_content_path ${dataDir}/static_content
+      css_path            ${dataDir}/static_content/css
+      pictures_path       ${dataDir}/static_content/pictures
+      mhonarc             ${pkgs.perlPackages.MHonArc}/bin/mhonarc
+    ''}
+
+    ${cfg.extraConfig}
+  '';
+
+  robotConfig = fqdn: domain: pkgs.writeText "${fqdn}-robot.conf" ''
+    ${optionalString (cfg.web.enable && domain.webHost != null) ''
+      # WEB
+      wwsympa_url         https://${domain.webHost}${strings.removeSuffix "/" domain.webLocation}
+    ''}
+
+    ${domain.extraConfig}
+  '';
+
+  transport = pkgs.writeText "transport.sympa" (concatStringsSep "\n" (flip map fqdns (domain: ''
+    ${domain}                        error:User unknown in recipient table
+    sympa@${domain}                  sympa:sympa@${domain}
+    listmaster@${domain}             sympa:listmaster@${domain}
+    bounce@${domain}                 sympabounce:sympa@${domain}
+    abuse-feedback-report@${domain}  sympabounce:sympa@${domain}
+  '')));
+
+  virtual = pkgs.writeText "virtual.sympa" (concatStringsSep "\n" (flip map fqdns (domain: ''
+    sympa-request@${domain}  postmaster@localhost
+    sympa-owner@${domain}    postmaster@localhost
+  '')));
+
+  listAliases = pkgs.writeText "list_aliases.tt2" ''
+    #--- [% list.name %]@[% list.domain %]: list transport map created at [% date %]
+    [% list.name %]@[% list.domain %] sympa:[% list.name %]@[% list.domain %]
+    [% list.name %]-request@[% list.domain %] sympa:[% list.name %]-request@[% list.domain %]
+    [% list.name %]-editor@[% list.domain %] sympa:[% list.name %]-editor@[% list.domain %]
+    #[% list.name %]-subscribe@[% list.domain %] sympa:[% list.name %]-subscribe@[%list.domain %]
+    [% list.name %]-unsubscribe@[% list.domain %] sympa:[% list.name %]-unsubscribe@[% list.domain %]
+    [% list.name %][% return_path_suffix %]@[% list.domain %] sympabounce:[% list.name %]@[% list.domain %]
+  '';
+in
+{
+
+  ###### interface
+
+  options = {
+
+    services.sympa = {
```

Comment:
A lot shorter would be `options.services.sympa = with types; {` which would also allow you to drop the `types.` in the entire block.